server/src/main/java/org/candlepin/policy/activationkey/ActivationKeyRules.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2009 - 2012 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.policy.activationkey;
+
+import org.candlepin.model.ConsumerType;
+import org.candlepin.model.Pool;
+import org.candlepin.model.activationkeys.ActivationKey;
+import org.candlepin.model.activationkeys.ActivationKeyPool;
+import org.candlepin.policy.ValidationResult;
+
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xnap.commons.i18n.I18n;
+
+import java.util.Optional;
+
+/**
+ * ActivationKeyRules
+ *
+ * Enforces rules similar to EntitlementRules.java for
+ * attaching pools to activation keys
+ */
+public class ActivationKeyRules {
+
+    private static Logger log = LoggerFactory.getLogger(ActivationKeyRules.class);
+
+    private I18n i18n;
+
+    @Inject
+    public ActivationKeyRules(I18n i18n) {
+        this.i18n = i18n;
+    }
+
+    /**
+     * Checks that the specified pool can be attached to the specified activation key. A result object
+     * is returned which contains errors/warnings, or is empty if validation did not find any issues.
+     *
+     * @param key The activation key to attach the pool to
+     *
+     * @param pool The pool to be attached to the activation key
+     *
+     * @param quantity The quantity of the pool the key will consume
+     *
+     * @return a result object which contains errors or warnings produced by the validation process.
+     *  In case the validation was successful, it will be empty.
+     */
+    public ValidationResult runPoolValidationForActivationKey(ActivationKey key, Pool pool, Long quantity) {
+        ValidationResult result = new ValidationResult();
+        if (quantity != null && quantity < 1) {
+            result.addError("rulefailed.invalid.quantity");
+        }
+
+        if (!"yes".equalsIgnoreCase(pool.getMergedProductAttribute(Pool.Attributes.MULTI_ENTITLEMENT))) {
```

Comment:
I wonder if we shouldn't transition these magic strings to our stringified-boolean check that covers many affirmative string values (i.e. "yes", "y", "true", "t", "1", etc.). Such a thing would give us flexibility to standardize those values at some undetermined point in the future and remove a magic string from the code.

Reply:
Fixed.